src/mjolnir/landmark_builder.cc

```diff
@@ -183,10 +184,58 @@ void LandmarkDatabase::insert_landmark(const std::string& name,
   pimpl->vacuum_analyze = true;
 }
 
-std::vector<Landmark> LandmarkDatabase::get_landmarks_in_bounding_box(const double minLat,
-                                                                      const double minLong,
-                                                                      const double maxLat,
-                                                                      const double maxLong) {
+// get multiple landmarks by their ids
+std::vector<Landmark> LandmarkDatabase::get_landmarks_by_ids(const std::vector<int64_t>& pkeys) {
+  sqlite3_stmt* get_landmarks_by_ids_stmt = nullptr;
+
+  // prepare the sql statement with appropriate number of question marks for the ids
+  std::string sql = "SELECT id, name, type, X(geom), Y(geom) FROM landmarks WHERE id IN (";
+  for (size_t i = 0; i < pkeys.size(); ++i) {
+    if (i > 0) {
+      sql += ", ";
+    }
+    sql += "?";
```

Comment:
we can put the actual value here instead of `?` and omit preparing the statement

Reply:
done. but i notice sqlite officially suggests to use preparing statement when query includes variables.

Reply:
Likely because they’re doing some internal validation of the variable type or so. I think in our use case here it’s fine to do as it is now, it’s not a multi-variable statement, just the same many times.